pkg/config/types/v1alpha1/v1alpha1_types.go

```diff
@@ -0,0 +1,515 @@
+	//   description: |
+	//     The port that the API server listens on internally.
+	//     This may be different than the port portion listed in the endpoint field above.
+	//     The default is 6443.
+	LocalAPIServerPort int `yaml:"localAPIServerPort,omitempty"`
+}
+
+// APIServerConfig represents kube apiserver config vals.
+type APIServerConfig struct {
+	//   description: |
+	//     The container image used in the API server manifest.
+	Image string `yaml:"image,omitempty"`
+	//   description: |
+	//     Extra arguments to supply to the API server.
+	ExtraArgs map[string]string `yaml:"extraArgs,omitempty"`
+	//   description: |
+	//     Extra certificate subject alternative names for the API server's certificate.
+	CertSANs []string `yaml:"certSANs,omitempty"`
+}
+
+// ControllerManagerConfig represents kube controller manager config vals.
+type ControllerManagerConfig struct {
+	//   description: |
+	//     The container image used in the controller manager manifest.
+	Image string `yaml:"image,omitempty"`
+	//   description: |
+	//     Extra arguments to supply to the controller manager.
+	ExtraArgs map[string]string `yaml:"extraArgs,omitempty"`
+}
+
+// SchedulerConfig represents kube scheduler config vals.
+type SchedulerConfig struct {
+	//   description: |
+	//     The container image used in the scheduler manifest.
+	Image string `yaml:"image,omitempty"`
+	//   description: |
+	//     Extra arguments to supply to the scheduler.
+	ExtraArgs map[string]string `yaml:"extraArgs,omitempty"`
+}
+
+// EtcdConfig represents etcd config vals.
+type EtcdConfig struct {
+	//   description: |
+	//     The container image used to create the etcd service.
+	ContainerImage string `yaml:"image,omitempty"`
+	//   description: |
+	//     The `ca` is the root certificate authority of the PKI.
+	//     It is composed of a base64 encoded `crt` and `key`.
+	//   examples:
+	//     - |
+	//       ca:
+	//         crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJIekNCMHF...
+	//         key: LS0tLS1CRUdJTiBFRDI1NTE5IFBSSVZBVEUgS0VZLS0tLS0KTUM...
+	RootCA *x509.PEMEncodedCertificateAndKey `yaml:"ca"`
+}
+
+// ClusterNetworkConfig represents kube networking config vals.
+type ClusterNetworkConfig struct {
+	//   description: |
+	//     The CNI used.
+	//   values:
+	//     - flannel
+	CNI string `yaml:"cni"`
+	//   description: |
+	//     The domain used by Kubernetes DNS.
+	//     The default is `cluster.local`
+	//   examples:
+	//     - cluser.local
+	DNSDomain string `yaml:"dnsDomain"`
+	//   description: |
+	//     The pod subnet CIDR.
+	//   examples:
+	//     -  |
+	//       podSubnets:
+	//         - 10.244.0.0/16
+	PodSubnet []string `yaml:"podSubnets"`
+	//   description: |
+	//     The service subnet CIDR.
+	//   examples:
+	//     -  |
+	//       serviceSubnets:
+	//         - 10.96.0.0/12
+	ServiceSubnet []string `yaml:"serviceSubnets"`
+}
+
+// Bond contains the various options for configuring a bonded interface.
```

Comment:
We probably should drop this for now while we don't actually support bonding.

Reply:
I think I want to add functionality to `docgen` that would allow skipping a field so we can keep the definition, but not render it.